### PR TITLE
Jerrycan

### PR DESCRIPTION
**Pull request by TiviPlus**
## About The Pull Request

Tee hee
https://streamable.com/jc9fer

## Why It's Good For The Game

Its fun
## Changelog
:cl:
add: Jerry cans can now be used to douse people and tiles in fuel to light them with flamers or lighters. Available in the req operations vendor
/:cl:

### CODE REVIEW

**Comment by AbsoluteJerry** [comment]
I love this! And hate this

**Comment by Hughgent** [comment]
It's clever, well done, and shouldn't be in requisitions.

litter these around ground side maps instead. TGMC is too 'future' to rely on plain old petrochemicals. But colonies which rely on older more proven technology would very much so use these.

**Comment by TiviPlus** [comment]
Its easier to balance for if its in req and its more attainable for the standard marine, and scattering them on-planet leads to xenos just melting them before marines land. And making them not meltable is a daft solution that I really dont approve of

**Comment by psykzz** [comment]
You could just litter them around the dropship equipment.
I agree being in req feels weird, especially in the Operational Supplies vendor.

These should be fluff items anyway, you should be able to do the same thing with a normal bucket.

**Comment by AbsoluteJerry** [comment]
TM when?

File: code/game/objects/items/reagent_containers/jerrycan.dm [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+		var/mob/living/M = A
+		if(!reagents.total_volume)
+			to_chat(user, "<span class='warning'>Theres no fuel left in [src]!</span>")
+			return
+		M.adjust_fire_stacks(10)
+		reagents.remove_reagent(/datum/reagent/fuel, fuel_usage)
+		to_chat(A, "<span class='warning'>[user] drenches you in fuel from [src]!<span>")
+		log_attack("[user] has doused [M] in fuel")
+
+	else if(istype(A, /obj/structure/reagent_dispensers/fueltank))
```
Comment (Rohesie) [inline code/game/objects/items/reagent_containers/jerrycan.dm:33]: Put this in `attack_obj()` instead

File: code/game/objects/items/reagent_containers/jerrycan.dm [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+	if(isturf(A))
+		if(A.density)
+			return
+		if(!reagents.total_volume)
+			to_chat(user, "<span class='warning'>Theres no fuel left in [src]!</span>")
+			return
+		new /obj/effect/decal/cleanable/liquid_fuel(A, fuel_usage/2)
+		reagents.remove_reagent(/datum/reagent/fuel, fuel_usage)
+		user.visible_message("<span class='notice'>[user] splashes some fuel on \the [A]</span>", "<span class='notice'>You splash some fuel on [A]</span>")
+	else if(isliving(A))
```
Comment (Rohesie) [inline code/game/objects/items/reagent_containers/jerrycan.dm:23]: You can use `attack()` for this one

File: code/game/objects/items/reagent_containers/jerrycan.dm [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+	icon_state = "canister"
+	volume = 200
+	list_reagents = list(/datum/reagent/fuel = 200)
+	///how much fuel we use up per spill
+	var/fuel_usage = 10
+
+/obj/item/reagent_containers/jerrycan/afterattack(atom/A, mob/user)
+	if(!(get_dist(user,A) <= 1))
+		return
+	if(isturf(A))
```
Comment (Rohesie) [inline code/game/objects/items/reagent_containers/jerrycan.dm:14]: If you could add an `attack_turf()` proc call to `/turf/attackby()`, and place this inside of it for your object, that'd be ideal

**Pull request merged**